src/v/kafka/server/handlers/topics/topic_utils.cc

```diff
@@ -79,7 +79,11 @@ ss::future<> wait_for_topics(
                    })
             .then([&md_cache, &results, timeout]() {
                 return wait_for_leaders(md_cache, results, timeout)
-                  .discard_result();
+                  .discard_result()
+                  .handle_exception_type([](const ss::timed_out_error&) {
+                      // discard timed out exception, even tho waiting failed
```

Comment:
How do we know for sure that the topic is created? Usually timeout is indicative error and it may mean either way: operation passed but timed out or operation failed and timed out.

Reply:
this operation (i.e. waiting for leaders) is executed after topic creation succeeded, hence we know that the topic already exists. here we only ignore the timeout returned by waiting for leaders